crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -14,7 +14,72 @@
 # limitations under the License.
 #
 
+require "open3"
+
 module Api
-  class Crowbar
+  class Crowbar < Tableless
+    attr_reader :version
+
+    def initialize
+      @version = ENV["CROWBAR_VERSION"]
+    end
+
+    def status
+      {
+        version: @version
+      }
+    end
+
+    def upgrade
+      status.merge!(
+        upgrade: {
+          upgrading: upgrading,
+          success: success,
+          failed: failed
+        }
+      )
+    end
+
+    def upgrade!
+      upgrade_script = Rails.root.join("..", "bin", "upgrade_admin_server.sh")
+
+      if upgrade_script.exist?
+        pid = spawn("sudo #{upgrade_script}")
+        Process.detach(pid)
+        Rails.logger.info("#{upgrade_script} executed with pid: #{pid}")
+
+        true
+      else
+        msg = "Could not find #{upgrade_script}"
+        Rails.logger.error(msg)
+        errors.add(:upgrade, msg)
+
+        false
+      end
+    end
+
+    def maintenance_updates_installed?
```

Comment:
This needs some more work, I assume. Specific return values based on the various exit status of "zypper patch-check", logging, etc.